app/controllers/api/v1/topics_controller.rb

```diff
@@ -0,0 +1,12 @@
+module Api
+  module V1
+    class TopicsController < ApplicationController
+      before_action :authenticate_user!
```

Comment:
This should be on the `ApplicationController` what you want is to skip it for some actions since most would be authenticated routes